pkg/controller/garbagecollector/error_recorder.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package garbagecollector
+
+import (
+	"sync"
+	"time"
+)
+
+// errorRecord represents an error with additional metadata
+type errorRecord struct {
```

Comment:
I'd prefer to use status throughout this entire thing, to differentiate from errors.

Reply:
Do we want to really rename everything? Eg. errorRecorder -> statusRecorder, errorRecord -> statusRecord, attemptToDeleteErrors -> attemptToDeleteStatuses ? I think this naming would give the struct higher responsibility than necessary, since the final status might comprise of other things than errors in the future.

I think the code would be more semantically closer as well, status seems too high level to me in the GC.
`gc.errorRecorder.SetAttemptToDeleteError(n, err)`  vs `gc.statusRecorder.SetAttemptToDeleteStatus(n, err)`

Reply:
IMO, status has a broader usage and I'm expecting we might be adding extra information there in the future :)